### PR TITLE
Github plugin: support branches with forward slashes

### PR DESCRIPTION
**Pull request by danielbeardsley**
The ref property contains a branch name and branch names can
contain forward slashes. Extract everything after the first two
slashes and use that as the prefix, while everything after is
treated as the branch name.

### CODE REVIEW

**Comment by jarstelfox** [comment]
CR 🔭

File: plugins/github.js [hunk 1]
```diff
@@ -26,13 +26,20 @@ exports.init = function(config, cimpler) {
 function extractBuildInfo(requestBody) {
    var info = JSON.parse(requestBody.payload);
 
-   // Filter out notifications about annotated tags
-   if (info.ref.indexOf('refs/tags/') == 0) {
+   // ref: "refs/heads/some-long-branch-name/maybe-even-slashes"
+   var matches = info.ref.match(/^(refs\/[^\/]+)\/(.*$)/);
```
Comment (djmetzle) [inline plugins/github.js]:
```suggestion
   const matches = info.ref.match(/^(refs\/[^\/]+)\/(.*$)/);
```

File: plugins/github.js [hunk 2]
```diff
@@ -26,13 +26,20 @@ exports.init = function(config, cimpler) {
...
+   if (!matches) {
+      return null;
+   }
+
+   var headType = matches[1];
```
Comment (djmetzle) [inline plugins/github.js]:
```suggestion
   const headType = matches[1];
```

File: plugins/github.js [hunk 3]
```diff
@@ -26,13 +26,20 @@ exports.init = function(config, cimpler) {
...
+   var branch = matches[2];
```
Comment (djmetzle) [inline plugins/github.js]:
```suggestion
   const branch = matches[2];
```

**Comment by djmetzle** [comment]
CR :factory: even without the `const` correctness...

Though that test has some _gnarly_ whitespace inconsistencies. Nothing to block on...

**Comment by djmetzle** [comment]
More CR :gear:

**Pull request merged**